### PR TITLE
MOB-298

### PR DESCRIPTION
**Pull request by shekhardahore**
Changelog
1. LiClientManage is now a property of LiSDKManager and thus an instance of LiSDKManager is now required to make API calls.
2. LiSDKManger now takes the LiAppCredentials object as a parameter of setup method, instead of taking them from info.plist.

### CODE REVIEW

File: LiCoreSDK/Managers/LiSDKManager.swift [hunk 1]
```diff
@@ -16,50 +16,41 @@ import Foundation
...
     }()
-    var visitorId: String? {
-        return UserDefaults.standard.string(forKey: LiCoreSDKConstants.LiUserDefaultConstants.liVisitorId)
+    /**
+     Use this function to congfigure community credentials.
+     Note: Call this function before using any SDK functionality. Preferably in AppDelegate on app launch.
+     
+     - parameter credentials: LiAppCredentials object
+    */
+    public class func setup(credentials: LiAppCredentials) {
```
Comment (AatishMolasi) [inline LiCoreSDK/Managers/LiSDKManager.swift]: Where is this method being called from?

Comment (shekhardahore) [reply LiCoreSDK/Managers/LiSDKManager.swift]: Supposed to be called from appdelegate.

File: LiCoreSDK/Auth/LiAppCredentials.swift [hunk 2]
```diff
@@ -40,40 +37,44 @@ struct LiAppCredentials {
...
-            print("SDK initalization failed: LiTenantId is missing.")
+            fatalError("SDK initalization failed: LiTenantId is missing.")
         }
         if apiProxyHost == "" {
-            print("SDK initalization failed: LiApiProxyHost is missing.")
+            fatalError("SDK initalization failed: LiApiProxyHost is missing.")
         }
         if clientAppName == "" {
-            print("SDK initalization failed: LiClientAppName is missing.")
+            fatalError("SDK initalization failed: LiClientAppName is missing.")
```
Comment (adityasharat) [inline LiCoreSDK/Auth/LiAppCredentials.swift]: I remember a util method to do this check. If there isn't one please make a utility method and make the code drier.

File: LiCoreSDK/Auth/LiAppCredentials.swift [hunk 3]
```diff
@@ -86,6 +87,7 @@ struct LiAppCredentials {
     }
     /**
      Funtion to return URLRequest that initiates Lithium login.
+     
      - returns: URLRequest containing the url and parameters for login.
  */
```
Comment (adityasharat) [inline LiCoreSDK/Auth/LiAppCredentials.swift]: please fix indentation

File: LiCoreSDK/Managers/LiSDKManager.swift [hunk 4]
```diff
@@ -16,50 +16,41 @@ import Foundation
...
-            } catch let error {
-                print("SDK initalization failed: \(error)")
-            }
+    /**
+     Returns the sharedInstance for the LiSDKManager singleton.
+     Note: This will throw a fatalError if this function is called before calling `setup`.
+    */
+    public static func shared() -> LiSDKManager {
+        if LiSDKManager.sharedInstance == nil {
+            fatalError("Please call setup function before using this class.")
```
Comment (adityasharat) [inline LiCoreSDK/Managers/LiSDKManager.swift]: Nice!

File: LiCoreSDK/Models/LiClientRequestParams.swift [hunk 5]
```diff
@@ -355,7 +355,7 @@ public struct LiDeviceIdFetchClientRequestParams: LiClientRequestParams {
         self.pushNotificationProvider = pushNotificationProvider.rawValue
     }
     public func getPostParams() -> [String: Any] {
-        let params: [String: Any] = ["type": LiQueryConstant.ResponseType.liUserDeviceIdFetchType, "device_id": deviceId, "client_id": LiSDKManager.sharedInstance.liAppCredentials.clientId, "push_notification_provider": pushNotificationProvider, "application_type": LiQueryConstant.ResponseType.liApplicationType]
+        let params: [String: Any] = ["type": LiQueryConstant.ResponseType.liUserDeviceIdFetchType, "device_id": deviceId, "client_id": LiSDKManager.shared().liAppCredentials.clientId, "push_notification_provider": pushNotificationProvider, "application_type": LiQueryConstant.ResponseType.liApplicationType]
```
Comment (adityasharat) [inline LiCoreSDK/Models/LiClientRequestParams.swift]: move `"type"`, `"device_id"` to constants class

File: LiCoreSDK/Managers/LiSDKManager.swift [hunk 6]
```diff
@@ -16,50 +16,41 @@ import Foundation
 public final class LiSDKManager {
     //Mark: - Internal
     var liAppCredentials: LiAppCredentials
+    private static var sharedInstance: LiSDKManager!
+    var visitorId: String? {
+        return UserDefaults.standard.string(forKey: LiCoreSDKConstants.LiUserDefaultConstants.liVisitorId)
```
Comment (adityasharat) [inline LiCoreSDK/Managers/LiSDKManager.swift]: Kind of prefer that constants are snake case. (All caps separated by _)
@shekhardahore What do you think?

**Pull request closed without merging**